### PR TITLE
Add RestClientResponseExceptionMapperTest test

### PR DESCRIPTION
**Pull request by gastaldi**

### CODE REVIEW

File: extensions/rest-client/deployment/src/test/java/io/quarkus/restclient/RestClientResponseExceptionMapperTest.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    @Inject
+    @RestClient
+    Client client;
+
+    @Test
+    public void testExceptionMapperIsCalled() {
+        assertThatExceptionOfType(IllegalArgumentException.class).isThrownBy(client::test);
+    }
+
+    @RegisterRestClient(baseUri = "http://localhost:8081")
```
Comment (geoand) [inline extensions/rest-client/deployment/src/test/java/io/quarkus/restclient/RestClientResponseExceptionMapperTest.java]: Is there perhaps a chance to get the port from the environment? 
I vaguely recall that being done somewhere...

**Comment by kenfinnigan** [comment]
What's the purpose of the test?

It's not related to a fix.

**Comment by gastaldi** [comment]
It's not a fix, It's more about showing that the ResponseExceptionMapper is called correctly, but I don't mind closing it, I just created the test because I wanted to check whether ResponseExceptionMappers were registered correctly

File: extensions/rest-client/deployment/src/test/java/io/quarkus/restclient/RestClientResponseExceptionMapperTest.java (hunk 1 above)
Comment (gastaldi) [reply extensions/rest-client/deployment/src/test/java/io/quarkus/restclient/RestClientResponseExceptionMapperTest.java]: I couldn't find another way to do that.

**Comment by kenfinnigan** [comment]
Is the situation covered by the TCK: https://github.com/eclipse/microprofile-rest-client/blob/master/tck/src/main/java/org/eclipse/microprofile/rest/client/tck/ExceptionMapperTest.java

If it isn't, then I'd recommend a TCK test instead

**Comment by gastaldi** [comment]
I think the only difference is that it uses an injected @RestClient. I agree it should be part of the TCK. I'll close it for now

**Pull request closed without merging**